src/libraries/System.Memory/tests/MemoryMarshal/CreateReadOnlySpanFromNullTerminated.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+using System.Runtime.InteropServices;
+using System.Buffers;
+
+namespace System.SpanTests
+{
+    public static partial class MemoryMarshalTests
+    {
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(256)]
+        public static unsafe void CreateReadOnlySpanFromNullTerminated_Char(int expectedLength)
+        {
+            using BoundedMemory<char> data = BoundedMemory.Allocate<char>(expectedLength + 1);
+            data.Span.Fill('s');
+            data.Span[^1] = '\0';
+            data.MakeReadonly();
+
+            fixed (char* expectedPtr = data.Span)
+            {
+                ReadOnlySpan<char> actual = MemoryMarshal.CreateReadOnlySpanFromNullTerminated(expectedPtr);
```

Comment:
What happens if we call `CreateReadOnlySpanFromNullTerminated` on a ptr that isn't null-terminated? Should we add a test for that?

Reply:
Then it'll either AV/seg fault or stop when it gets to int.MaxValue characters.  If the former, crash, and if the latter, it'll throw an exception.  Testing the latter would require a 2GB allocation, which seems excessive.

Reply:
I see. I am trying to understand the behavior here:
https://source.dot.net/#System.Private.CoreLib/String.cs,626

Are you suggesting this isn't something we should document as part of the API contract? Is the fact that we throw after the call to `IndexOf` non-deterministic? Or is it something that is an implementation detail which is not part of the API contract and could change, and hence callers shouldn't rely on it?

> Testing the latter would require a 2GB allocation, which seems excessive.

Right, agreed. That makes sense.

Reply:
> Are you suggesting this isn't something we should document as part of the API contract?

If you're passing in a non-null terminated ptr to a marshaling method, you're already violating the contract.  At that point, we're going to walk through arbitrary memory.  It might happen to find a random 0 in memory, or it might AV, or it might go to the grocery store and buy milk... who knows.